### PR TITLE
migrate to typescript/work in windows env/add missing translations

### PR DESCRIPTION
**Pull request by MohammedFaragallah**
If you are interested in this I'm happy to make any required changes

### CODE REVIEW

**Comment by developerium** [comment]
I don't have a react admin setup, are you guys sure about these changes? @MohammedFaragallah  @ahmadi-akbar

**Comment by MohammedFaragallah** [comment]
positive @developerium I'm using these changes in production

**Comment by developerium** [comment]
Could you resolve this conflict on `index.js` file? @MohammedFaragallah

**Comment by MohammedFaragallah** [comment]
index.js should not be edited any more typescript build it automatically from index.ts

**Comment by ahmadi-akbar** [comment]
> index.js should not be edited any more typescript build it automatically from index.ts

hello, you add new translation to  "index.ts" file but  we have warning again,
i think we must edit 
main file in package.json to "main": "src/index.js"
or add translations to "lib/index.js"

## PS
### warnings :
image

### ra-language-arabic package.json:
image

**Comment by MohammedFaragallah** [comment]
@ahmadi-akbar sorry about that I'll submit a fix soon

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+"use strict";
```
Comment (ahmadi-akbar) [inline lib/index.js]: Please take look at  official translations 
https://github.com/marmelab/react-admin/tree/master/packages/ra-language-french
https://github.com/marmelab/react-admin/tree/master/packages/ra-language-english
these repository have only "src/index.ts" file, removed other files

thanks you

Comment (MohammedFaragallah) [reply lib/index.js]: The published npm package have these files

Can you please elaborate more on the problem here

Comment (ahmadi-akbar) [reply lib/index.js]: these files generated by "yarn build" script automatically

Comment (MohammedFaragallah) [reply lib/index.js]: I really don't understand the problem here, my changes are based on the link you're referring and everything works fine with me

**Comment by developerium** [comment]
Hey guys, this PR indicates that there is a conflict on `index.js` if you could resolve this (as I don't have a working react-admin setup and I trust you guys 😄 ), I'm happy to merge this PR
@MohammedFaragallah @ahmadi-akbar

**Comment by MohammedFaragallah** [comment]
@developerium all should be good now

**Comment by developerium** [comment]
Thanks @MohammedFaragallah @ahmadi-akbar

**Pull request merged**